activerecord/lib/active_record/sanitization.rb

```diff
@@ -36,8 +36,7 @@ def sanitize_sql_for_conditions(condition)
           else        condition
           end
         end
-        alias_method :sanitize_sql, :sanitize_sql_for_conditions
-        alias_method :sanitize_conditions, :sanitize_sql
```

Comment:
Yes, it is documented in http://api.rubyonrails.org/classes/ActiveRecord/Sanitization/ClassMethods.html#method-i-sanitize_conditions, so it is public API.